src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Linn.Stores.Domain.LinnApps.ImportBooks
+{
+    using System.Collections.Generic;
+    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Parts;
+
+    public class ImportBookService : IImportBookService
+    {
+        private readonly IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> InvoiceDetailRepository;
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> OrderDetailRepository;
+        private readonly IRepository<ImportBookPostEntry, ImportBookPostEntryKey> PostEntryRepository;
+
+
+        public ImportBookService(
+            IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> invoiceDetailRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<ImportBookPostEntry, ImportBookPostEntryKey> postEntryRepository)
+        {
+            this.InvoiceDetailRepository = invoiceDetailRepository;
+            this.OrderDetailRepository = orderDetailRepository;
+            this.PostEntryRepository = postEntryRepository;
+        }
+
+        public void Update(
+            ImportBook from,
+            IEnumerable<ImportBookInvoiceDetail> invoiceDetails,
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            IEnumerable<ImportBookPostEntry> postEntries)
+        {
+            this.UpdateInvoiceDetails(invoiceDetails);
+
+            this.UpdateOrderDetails(orderDetails);
+
+            this.UpdatePostEntries(postEntries);
+        }
+
+        public void UpdateInvoiceDetails(IEnumerable<ImportBookInvoiceDetail> details)
+        {
+            foreach (var detail in details)
+            {
+                var currentDetail = this.InvoiceDetailRepository.FindById(new ImportBookInvoiceDetailKey(detail.ImportBookId, detail.LineNumber));
```

Comment:
you don't need to look these up again. they should be on the from field above. Just update / add to that as appropriate